packages/apps/project/src/features/projectTaskSlice.js

```diff
@@ -80,17 +101,162 @@ export const fetchProjectTaskById = createAsyncThunk(
+      fetchFunction: _getProjectTaskTag,
+      data,
+      action: 'Project_SliceAction_GetProjectTaskTag',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const searchTargetVersion = createAsyncThunk(
+  'project_projectTask/searchTargetVersion',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _searchTargetVersion,
+      data,
+      action: 'Project_SliceAction_SearchTargetVersion',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const searchCategory = createAsyncThunk(
+  'project_projectTask/searchCategory',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _searchCategory,
+      data,
+      action: 'Project_SliceAction_SearchCategory',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const searchSection = createAsyncThunk(
+  'project_projectTask/searchSection',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _searchSection,
+      data,
+      action: 'Project_SliceAction_SearchSection',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const searchPriority = createAsyncThunk(
+  'project_projectTask/searchPriority',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _searchPriority,
+      data,
+      action: 'Project_SliceAction_SearchPriority',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const searchStatus = createAsyncThunk(
+  'project_projectTask/searchStatus',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _searchStatus,
+      data,
+      action: 'Project_SliceAction_SearchStatus',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    });
+  },
+);
+
+export const updateProjectTask = createAsyncThunk(
+  'project_projectTask/updateProjectTask',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _updateProjectTask,
+      data,
+      action: 'Project_SliceAction_UpdateProjectTask',
+      getState,
+      responseOptions: {isArrayResponse: true},
+    }).then(() => {
+      return handlerApiCall({
+        fetchFunction: _fetchProjectTaskById,
+        data: {projecTaskId: data?.projectTask?.id},
+        action: 'Project_SliceAction_FetchProjectTaskById',
+        getState,
+        responseOptions: {isArrayResponse: false},
+      });
+    });
+  },
+);
+
+export const createProjectTask = createAsyncThunk(
+  'project_projectTask/createProjectTask',
+  async function (data, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _updateProjectTask,
+      data,
+      action: 'Project_SliceAction_CreateProjectTask',
+      getState,
+      responseOptions: {isArrayResponse: false},
+    });
+  },
+);
```

Comment:
Since you use the same fetchFunction as updateProjectTask, I think you can remove createProjectTask and do all in updateProjectTask with condition (if isCreation is present). If you do that, rename updateProjectTask